SSA/Projects/InstCombine/HackersDelight.lean

```diff
@@ -9,27 +9,55 @@ namespace HackersDelight
 
 namespace Ch2Basics
 
-theorem neg_add_eq_neg_sub {x: BitVec w } :
-    -(x + y) = -x - y := by
-  ring
-
-theorem neg_eq_not_add_one {x : BitVec w} :
-    -x = ~~~x + 1 := by
-  exact BitVec.neg_eq_not_add x
-
 theorem not_eq_neg_sub_one {x : BitVec w} :
     ~~~ x = -x - 1 := by
   apply BitVec.eq_sub_iff_add_eq.mpr
   rw [BitVec.neg_eq_not_add]
 
+theorem not_and_eq_not_or_not {x y : BitVec w} :
```

Comment:
It's up to you stylistically, but you could lift out the implicit variable declaration

```suggestion
variable {x y : BitVec w}
theorem not_and_eq_not_or_not :
```